pkg/query/storeset.go

```diff
@@ -252,7 +255,11 @@ func (s *StoreSet) Update(ctx context.Context) {
 		// Note: No external labels means strictly store gateway or ruler and it is fine to have access to multiple instances of them.
 		// Any other component will error out if it will be configured with empty external labels.
 		externalLabels := externalLabelsFromStore(store)
-		if len(store.LabelSets()) > 0 && externalLabelOccurrencesInStores[externalLabels] != 1 {
+		if len(store.LabelSets()) > 0 &&
+			store.storeType != nil &&
+			(store.storeType.ToProto() == storepb.StoreType_QUERY ||
```

Comment:
Can we just exclude `StoreType_STORE`? It will be much safe. It's the only thing that we change here - store GW handing, right?